### PR TITLE
Change 'allowed' rule to use containers instead of lists.

### PR DESCRIPTION
**Pull request by crunk1**
Addresses #384.

### CODE REVIEW

**Comment by funkyfuture** [comment]
lptm.

**Comment by funkyfuture** [comment]
you could actually add a test with your `Enum` use-case for Python 3.4+.

**Comment by crunk1** [comment]
For simplicity, I don't think it is necessary to build out my entire Enum behavior. I'll keep it to testing a Container type other than list, ie I'll use a set.

By the way, I tried to modify the 'allowed' documentation, but I don't know where the Sphinx gloasary is. I wanted to add :term:`container`.

**Comment by funkyfuture** [comment]
the [Python docs glossary](https://docs.python.org/3/glossary.html) does not include the term *container*. you may point to:

    :class:`py3:collectionsabc.Container`

**Comment by crunk1** [comment]
Done.

File: docs/validation-rules.rst [hunk 1]
```diff
@@ -13,7 +13,7 @@ For a full elaboration refer to :ref:`this paragraph <allowing-the-unknown>`.
 
 allowed
 -------
-This rule takes a :term:`container` of allowed values.
+This rule takes a :class:`py3:collectionsabc.Container` of allowed values.
```
Comment (funkyfuture) [inline docs/validation-rules.rst]: typo: `collections.abc`

**Pull request merged**